engine.py

```diff
@@ -23,200 +19,240 @@
 import nukescripts
 
 class NukeEngine(tank.platform.Engine):
+    """
+    An engine that supports Nuke 6.3v5+ and Hiero 9.0v*+.
+    """
+    # Define the different areas where menu events can occur in Hiero.
+    (HIERO_BIN_AREA, HIERO_SPREADSHEET_AREA, HIERO_TIMELINE_AREA) = range(3)
 
-    ##########################################################################################
-    # init
+    def __init__(self, *args, **kwargs):
+        self._hiero = nuke.env.get("hiero")
```

Comment:
Wait - `has_ui` is part of the engine base class and is used all over the place. That needs to remain the same (see https://support.shotgunsoftware.com/entries/95441127#has_ui%20property abd https://github.com/shotgunsoftware/tk-core/blob/master/python/tank/platform/engine.py#L383 )

We also need to be backwards compatible with these kinds of changes :)